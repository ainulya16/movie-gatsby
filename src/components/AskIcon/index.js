import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  display: inline-block;
  background: #596c80;
  color: white;
  font-weight: bolder;
  width: 20px;
  height: 20px;
  line-height: 20px;
  border-radius: 10px;
`;
const AskIcon = props => {
  const { containerStyle, textStyle, ...attr } = props;
  return (
    <Container style={containerStyle} {...attr}>
      <span style={textStyle}>?</span>
    </Container>
  );
};

AskIcon.propTypes = {
  containerStyle: PropTypes.shape(),
  textStyle: PropTypes.shape(),
};

AskIcon.defaultProps = {
  containerStyle: {},
  textStyle: {},
};

export default AskIcon;
