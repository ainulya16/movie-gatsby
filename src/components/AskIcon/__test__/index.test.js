import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import AskIcon from '..';

describe('Ask Icon', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<AskIcon />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should have text "?"', () => {
    const wrapper = shallow(<AskIcon />);
    expect(wrapper.text()).toEqual('?');
  });
});
