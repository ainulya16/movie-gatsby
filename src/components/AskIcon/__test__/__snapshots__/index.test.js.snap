// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ask Icon should match snapshot 1`] = `
.c0 {
  display: inline-block;
  background: #596c80;
  color: white;
  font-weight: bolder;
  width: 20px;
  height: 20px;
  line-height: 20px;
  border-radius: 10px;
}

<div
  className="c0"
  style={Object {}}
>
  <span
    style={Object {}}
  >
    ?
  </span>
</div>
`;
